blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/IncomingMetricMetadataAnalyzer.java

```diff
@@ -95,11 +93,16 @@ private IncomingMetricException checkMeta(Locator locator, String key, String in
         }
 
         List<IncomingMetricException> problems = new ArrayList<IncomingMetricException>();
+        IncomingMetricException typeProblem = null;
+        IncomingMetricException unitProblem = null;
 
-        IncomingMetricException typeProblem = checkMeta(metric.getLocator(), MetricMetadata.TYPE.name().toLowerCase(),
-                metric.getDataType().toString());
-        IncomingMetricException unitProblem = checkMeta(metric.getLocator(), MetricMetadata.UNIT.name().toLowerCase(),
-                metric.getUnit());
+        if (metric.getDataType() != DataType.NUMERIC) {
+            typeProblem = checkMeta(metric.getLocator(), MetricMetadata.TYPE.name().toLowerCase(),
+                    metric.getDataType().toString());
+
+            unitProblem = checkMeta(metric.getLocator(), MetricMetadata.UNIT.name().toLowerCase(),
```

Comment:
`checkMeta` for unit does not have to be in `metric.getDataType() != DataType.NUMERIC`, I think. It will be handled as a separate block of code in https://github.com/rackerlabs/blueflood/pull/432